### PR TITLE
[INFRA/RULES] Added Link pulling for pulling into and out of caches.

### PR DESCRIPTION
**Pull request by Neelesh99**
# Summary

Quick change, to ensure than when a Linked rule is elected into/out of the RulesInPlay that the rule's Link is also pulled into/ out of cache.

## Test Plan

- Test added

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/common/rules/globalrulescache.go [hunk 1]
```diff
@@ -53,6 +57,10 @@ func PullRuleOutOfPlay(rulename string) error {
 func pullRuleOutOfPlayInternal(rulename string, allRules map[string]RuleMatrix, playRules map[string]RuleMatrix) error {
 	if _, ok := allRules[rulename]; ok {
 		if _, ok := playRules[rulename]; ok {
+			linkRule, linked := checkLinking(rulename, allRules)
+			if linked {
+				delete(playRules, linkRule)
```
Comment (NGovani) [inline internal/common/rules/globalrulescache.go:62]: Is it possible for linkRule to have another linked rule which would have to either be added to play or removed?

**Review by NGovani — APPROVED** [review]
One minor comment but otherwise LGTM

File: internal/common/rules/globalrulescache.go (hunk 1 above)
Comment (Neelesh99) [reply internal/common/rules/globalrulescache.go:62]: I think from a design perspective this isn't possible, and I wasn't sure whether to recursively look and fail or just ignore like this.

**Pull request merged**